src/typeahead/typeahead-window.ts

```diff
@@ -73,7 +75,13 @@ export class NgbTypeaheadWindow {
 
   next() { this.activeIdx = (this.activeIdx + 1) % this.results.length; }
 
-  prev() { this.activeIdx = (this.activeIdx === 0 ? this.results.length - 1 : this.activeIdx - 1); }
+  prev() {
+    this.activeIdx =
+        ((this.autoSelectItem === true && this.activeIdx === 0) ||
+                 (this.autoSelectItem === false && this.activeIdx === -1) ?
+             this.results.length - 1 :
+             this.activeIdx - 1);
```

Comment:
There is way too much logic compacted in these lines - find a way to simplify readabiliity